comp/core/flare/endpoint.go

```diff
@@ -0,0 +1,63 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package flare
+
+import (
+	"encoding/json"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
+)
+
+// EndpointProvider wraps the flare component with a http.Handler interface
+type EndpointProvider struct {
+	flareComp *flare
+}
+
```

Comment:
Can you add a constructor for `EndpointProvider`, so adding a new dependencie will be easier?